py4syn/epics/QE65000Class.py

```diff
@@ -0,0 +1,262 @@
+
+    def getCountTime(self):
+        return self.pvTime.get()
+
+    def setCountStop(self):
+        # TODO: test
+        # Work only when in continuos mode
+        pass
+
+    # save the spectrum intensity in a mca file
+    # or an hdf file
+    def saveSpectrum(self, **kwargs):
+        self.spectrum = self.pvSpectrum.get(as_numpy=True)
+
+        # save a unique point
+        if self.image is None:
+            fileName = self.fileName
+            idx = 1
+            if(fileName):
+                prefix = fileName.split('.')[0]
+                while os.path.exists('%s_ocean_%04d.mca' % (prefix, idx)):
+                    idx += 1
+                fileName = '%s_ocean_%04d.mca' % (prefix, idx)
+                np.savetxt(fileName, self.spectrum, fmt='%f')
+        else:
+            # add a point on hdf file
+            self.col = int(self.lastPos/self.rows)
+            self.row = self.lastPos - self.rows*self.col
+            # if is an odd line
+            if (self.col % 2 != 0):
+                self.row = -1*(self.row+1)
+
+            self.image[self.col, self.row, :] = self.spectrum[:self.imageDeep]
+
+            self.lastPos += 1
+
+    def isCountRunning(self):
+        return (self.acquiring)
+
+    def wait(self):
+        """
+        Blocks until the acquisition completes.
+        """
+        if self.acquiring is False:
+            return
+
+        self.acquireChanged.clear()
+        # while acquiring and not time out waits
+        # TODO: find a better way to do this
+        while self.acquiring and self.timer.check():
+            self.acquireChanged.wait(0.001)
+            self.acquireChanged.clear()
+
+        if self.timer.expired():
+            raise RuntimeError('QE65000 is not answering')
+
+    def canMonitor(self):
+        """ Returns false indicating cannot be use as a counter monitor"""
+        return False
+
+    def canStopCount(self):
+        """
+        Returns true indicating that Dxp has a stop command.
+        """
+        return False
+
+    def getValue(self, **kwargs):
+        """Return intensity
+        It's a dummy method, always return 1.0. """
+        self.saveSpectrum()
+        return 1.0
+
+    def isCounting(self):
+        return self.acquiring
+
+    def startCount(self):
+        """ Starts acquiring an spectrum
+        It's necessary to call setCounTime before"""
+
+        if self.acquiring:
+            raise RuntimeError('Already counting')
+
+        self.acquiring = True
+        self.pvStart.put("Stop")
+        # resets initial time value
+        self.timer.mark()
+
+    def stopCount(self):
+        self.setCountStop()
+
+    def setPresetValue(self, channel, val):
+        """Dummy method"""
+        pass
+
+    # TODO: verificar
+    def close(self):
+        """Stops an ongoing acquisition, if any, and puts the EPICS IOC in
+        idle state."""
+        pass
+
+    def startCollectImage(self, rows=0, cols=0):
```

Comment:
Yes! In order to keep this code that is mainly the same but in one place. A base class or move this to something that can be imported and used by all of them and future developments with the same idea.